src/fwriteR.c

```diff
@@ -228,8 +228,9 @@ SEXP fwriteR(
   int firstListColumn = 0;
   for (int j=0; j<args.ncol; j++) {
     SEXP column = VECTOR_ELT(DFcoerced, j);
-    if (args.nrow != length(column))
-      error(_("Column %d's length (%d) is not the same as column 1's length (%d)"), j+1, length(column), args.nrow);
```

Comment:
Great spot too. Yes this could have been tripping up tests with a hard to track memory fault like you were seeing.